Use envvars NIX_CACHE_HOME, NIX_CONFIG_HOME, NIX_DATA_HOME, NIX_STATE_HOME if defined (#11351)
NixOS/nix: doc/manual/src/command-ref/env-common.md

- [`XDG_STATE_HOME`]{#env-XDG_STATE_HOME} (default `~/.local/state`)
- [`XDG_CACHE_HOME`]{#env-XDG_CACHE_HOME} (default `~/.cache`)

[XDG Base Directory Specification]: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
[`use-xdg-base-directories`]: @docroot@/command-ref/conf-file.md#conf-use-xdg-base-directories

> In addition, the following environment variables can be used to override the XDG base directories:

Reviewer: ```suggestion
In addition, setting the following environment variables overrides the XDG base directories:
```

- [`NIX_CONFIG_HOME`]{#env-NIX_CONFIG_HOME} (default `$XDG_CONFIG_HOME/nix`)
- [`NIX_STATE_HOME`]{#env-NIX_STATE_HOME} (default `$XDG_STATE_HOME/nix`)
- [`NIX_CACHE_HOME`]{#env-NIX_CACHE_HOME} (default `$XDG_CACHE_HOME/nix`)

For example, when [`use-xdg-base-directories`] is enabled, the configuration directory is:

1. `$NIX_CONFIG_HOME`, if it is defined
2. Otherwise, `$XDG_CONFIG_HOME/nix`, if `XDG_CONFIG_HOME` is defined
3. Otherwise, `~/.config/nix`.

> 

Reviewer: I considered framing this as an example (in which case I would have suggested highlighting it as such), but really this is the generic mechanism. There's nothing exemplary about this demonstration.

Author: Yes, I felt that it's an example, but only for 2 other cases (state and cache). Perhaps replace "For example" with "Specifically"? I feel that it's a bit better to say that this details the previous paragraph.

But if you feel it's best without any word, I'll happily revert.

Reviewer: We could say "Specifically, ..." or "Concretely, ...", but it seems rather superfluous, because what comes after is concrete and specific and I doubt it needs extra words to emphasize that.

Author: I'll just revert!

> So, when [`use-xdg-base-directories`] is enabled, the configuration directory is:
> 
> 1. `$NIX_CONFIG_HOME`, if it is defined
> 2. Otherwise, `$XDG_CONFIG_HOME/nix`, if `XDG_CONFIG_HOME` is defined
> 3. Otherwise, `~/.config/nix`.
> 
> And likewise for the state and cache directories.

Reviewer: ```suggestion
When [`use-xdg-base-directories`] is enabled, the configuration directory is:

1. `$NIX_CONFIG_HOME`, if it is defined
2. Otherwise, `$XDG_CONFIG_HOME/nix`, if `XDG_CONFIG_HOME` is defined
3. Otherwise, `~/.config/nix`.

Likewise for the state and cache directories.
```
